src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayload.cs

```diff
@@ -18,7 +18,8 @@ internal class CounterPayload : ICounterPayload
             string unit,
             double value,
             CounterType counterType,
-            float interval)
+            float interval,
+            Dictionary<string, string> metadata)
```

Comment:
I saw that the EventCounterTriggerTest was updated to pass `null` for this parameter. At least in testing, the `CounterPayload` class allocates a `Dictionary<string, string>` for the property and then tosses it out by setting the property to `null`. I think it might be good to using null-coalescing to keep the property consistently non-null:

```cs
Metadata = metadata ?? new Dictionary<string, string>(0)
```

This still allocates a Dictionary per counter payload. This can be optimized further by changing the `Metadata` property to be `IReadOnlyDictionary<string, string>` and null-coalesce will a statically created empty dictionary:

```cs
#if NETSTANDARD
private static readonly IReadOnlyDictionary<string, string> Empty = new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(0));
#else
private static readonly IReadOnlyDictionary<string, string> Empty = ImmutableDictionary<string, string>.Empty();
#endif

...

Metadata = metadata ?? Empty;
```

The preprocessor split might not be necessary if ImmutableDictionary is defined in `netstandard2.0` but I don't believe it is. This code is also manually written without compiler verification, so you might need to make adjustments.